src/query-performance-monitoring/validations/performance_metrics_validations.go

```diff
@@ -1,69 +1,72 @@
 package validations
 
 import (
-	"fmt"
-
 	"github.com/newrelic/infra-integrations-sdk/v3/log"
 	performancedbconnection "github.com/newrelic/nri-postgresql/src/connection"
 	commonutils "github.com/newrelic/nri-postgresql/src/query-performance-monitoring/common-utils"
 )
 
-func isExtensionEnabled(conn *performancedbconnection.PGSQLConnection, extensionName string) (bool, error) {
-	rows, err := conn.Queryx(fmt.Sprintf("SELECT count(*) FROM pg_extension WHERE extname = '%s'", extensionName))
+var extensions map[string]bool
+
+func fetchAllExtensions(conn *performancedbconnection.PGSQLConnection) error {
+	rows, err := conn.Queryx("SELECT extname FROM pg_extension")
 	if err != nil {
 		log.Error("Error executing query: ", err.Error())
-		return false, err
+		return err
 	}
 	defer rows.Close()
-
-	var count int
+	extensions = make(map[string]bool)
 	for rows.Next() {
-		if err := rows.Scan(&count); err != nil {
+		var extname string
+		if err := rows.Scan(&extname); err != nil {
 			log.Error("Error scanning rows: ", err.Error())
+			return err
 		}
+		extensions[extname] = true
 	}
-	if err := rows.Err(); err != nil {
-		log.Error(err.Error())
-	}
+	return nil
+}
 
-	return count > 0, nil
+func isExtensionEnabled(extensionName string) bool {
+	return extensions[extensionName]
 }
 
-func CheckSlowQueryMetricsFetchEligibility(conn *performancedbconnection.PGSQLConnection, version uint64) (bool, error) {
-	if version < commonutils.PostgresVersion12 {
-		return false, nil
-	}
-	return isExtensionEnabled(conn, "pg_stat_statements")
+func CheckSlowQueryMetricsFetchEligibility(conn *performancedbconnection.PGSQLConnection) (bool, error) {
+	loadExtensionsMap(conn)
+	return isExtensionEnabled("pg_stat_statements"), nil
 }
 
-func CheckWaitEventMetricsFetchEligibility(conn *performancedbconnection.PGSQLConnection, version uint64) (bool, error) {
-	if version < commonutils.PostgresVersion12 {
-		return false, nil
-	}
-	pgWaitExtension, waitErr := isExtensionEnabled(conn, "pg_wait_sampling")
-	if waitErr != nil {
-		return false, waitErr
-	}
-	pgStatExtension, statErr := isExtensionEnabled(conn, "pg_stat_statements")
-	if statErr != nil {
-		return false, statErr
-	}
-	return pgWaitExtension && pgStatExtension, nil
+func CheckWaitEventMetricsFetchEligibility(conn *performancedbconnection.PGSQLConnection) (bool, error) {
+	loadExtensionsMap(conn)
```

Comment:
this behaviour is quite obscure why are we doing this? at least some comments are needed